src/logtab.cpp

```diff
@@ -96,8 +96,34 @@ void LogTab::InitMenus()
     m_hideSelectedType = new QAction(QIcon(GetThemedIcon(":/ctx-hide.png")), "Hide all events of this type", this);
     connect(m_hideSelectedType, &QAction::triggered, this, &LogTab::RowHideSelectedType);
 
+    // Top-level `Export events`
     m_exportToTabAction = new QAction(QIcon(GetThemedIcon(":/ctx-newtab.png")), "Export event(s) to new tab", this);
-    connect(m_exportToTabAction, &QAction::triggered, this, &LogTab::ExportToNewTab);
+    connect(m_exportToTabAction, &QAction::triggered, [this]() { ExportToNewTab(); });
+    // Populate `Export to tab` menu
+    m_exportToTabMenu = new QMenu("Export to new tab",this);
+    m_exportToTabMenu->setIcon(QIcon(GetThemedIcon(":/ctx-newtab.png")));
+    {
+        // This thread
+        QAction* exportSelectedType = new QAction(QIcon(GetThemedIcon(":/ctx-newtab.png")), "this type", m_exportToTabMenu);
+        connect(exportSelectedType, &QAction::triggered, [this]() { ExportToNewTab(COL::Key); });
+        m_exportToTabMenu->addAction(exportSelectedType);
+        // This session
```

Comment:
I am not sure what you are referring to here.
Do you mean, e.g., disabling exporting a session, if the session is not set, i.e. "-"?
I would rather not do so, because it might be useful to export/highlight all events logged by background threads

Reply:
I mean some data structure with icon, text and column to operate on that we can iterate through to add all the items.

Reply:
I see. I changed the pull-request accordingly